cmd/orca-tasks/task_watcher.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2019, Google, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package orca_tasks
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/spinnaker/spin/cmd/gateclient"
+)
+
+// WaitForSuccessfulTask observes an Orca task to see if it completed successfully.
+func WaitForSuccessfulTask(gateClient *gateclient.GatewayClient, taskRef map[string]interface{}, maxAttempts int) error {
+	id := idFromTaskRef(taskRef)
+	task, resp, err := gateClient.TaskControllerApi.GetTaskUsingGET1(gateClient.Context, id)
+
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		return fmt.Errorf("Encountered an error saving application, status code: %d\n", resp.StatusCode)
+	}
+	if !taskSucceeded(task) {
+		return fmt.Errorf("Encountered an error saving application, task output was: %v\n", task)
+	}
+
+	attempts := 0
+	for (task == nil || !taskCompleted(task)) && attempts < maxAttempts {
+		id := idFromTaskRef(taskRef)
+		task, resp, err = gateClient.TaskControllerApi.GetTaskUsingGET1(gateClient.Context, id)
+		attempts += 1
+		time.Sleep(time.Duration(attempts*attempts) * time.Second)
```

Comment:
This is minor, but we'll always sleep for some amount of time *after* the last request, so we're artificially making things take longer than they need to.  We probably want this at the beginning of the loop (which also will add a delay between the poll at line 28 and the first poll in the loop.)

Reply:
Also, is this deliberately quadratic backoff?  I'm less concerned about the particular backoff algorithm, but the max wait time here is < 1 min for 5 attempts...is that enough?

Reply:
I think the number of attempts is something we'll have to tune. This is just a heuristic that's been sufficient for now.